ntoskrnl/ex/time.c

```diff
@@ -336,6 +370,22 @@ ExpSetTimeZoneInformation(PRTL_TIME_ZONE_INFORMATION TimeZoneInformation)
     TIME_FIELDS TimeFields;
     DPRINT("ExpSetTimeZoneInformation() called\n");
 
+    /* Get the Shared User Data System Time into the local SystemTime */
+    do
+    {
+        SystemTime.HighPart = SharedUserData->SystemTime.High1Time;
+        SystemTime.LowPart = SharedUserData->SystemTime.LowPart;
+    }
+    while (SystemTime.HighPart != SharedUserData->SystemTime.High2Time);
+
```

Comment:
Use KeQuerySystemTime instead.

Reply:
Revised to use KeQuerySystemTime. Thanks.